lib/ZwaveClient.ts

```diff
@@ -3352,11 +3354,9 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 	 * Emitted when we receive a node `notification` event
 	 *
 	 */
-	private _onNodeNotification(
-		zwaveNode: ZWaveNode,
-		ccId: CommandClasses,
-		args: Record<string, unknown>
-	) {
+	private _onNodeNotification: ZWaveNotificationCallback = (...parms) => {
+		const [zwaveNode, ccId, args] = parms
```

Comment:
This seems a bit weird, but destructuring here instead of having 3 separate arguments makes it so TS understands the connection between the parameter types and can deduce the type of `args` in the if branches.

Reply:
👍🏼  Thanks :)